src/commands/bundles.js

```diff
@@ -0,0 +1,160 @@
+const { authorize, sheets } = require('../sheets');
+
+const generateLocaleExports = (adapter, outputPath, locales) => {
+  const { index, indexName } = adapter;
+  const localeExportPath = `${outputPath}/${indexName}`;
+  const existingLocales = tryRequire(localeExportPath) || false;
+
+  if (existingLocales) {
+    const existingLocaleKeys = Object.keys(existingLocales).sort();
+
+    if (
+      locales.length === existingLocaleKeys.length &&
+      locales.sort().every((loc, idx) => loc === existingLocaleKeys[idx])
+    ) return;
+  }
+
+  const localeList = locales.map((locale) => [toSnakeCase(locale).toUpperCase(), locale]);
+  const i18nIndexFile = stripWhitespace(index({ list: localeList }));
+
+  return writeFile(localeExportPath, i18nIndexFile);
+}
+
+const generateBundleExports = (adapter, outputPath, files, locales) => {
+  const { index, indexName } = adapter;
+  if (!files || !files.length || files.every(f => !f.newFile)) return;
+
+  const bundleList = (files || []).map((file) => (
+    [file.bundleName, file.fileName]
+  ));
+
+  bundleList.sort((array) => array[0]);
+
+  const langIndexFile = stripWhitespace(index({ list: bundleList, mirrorExports: true }));
+
+  return locales.map((locale) => (
+    writeFile(`${outputPath}/${locale}/${indexName}`, langIndexFile)
+  ));
+}
+
+
+function transformBundle(ext, serializer, output, locale, bundles) {
+  return Object.keys(bundles).map(function (bundleName) {
+    const bundleFile = `${ bundleName }${ ext }`;
+    const bundlePath = `${ output }/${ locale }/${ bundleFile }`;
+    const bundle = bundles[bundleName];
+    const data = serializer(bundle) + '\n';
+
+    const existingBundle = tryRequire(bundlePath);
+
+    const bundleData = {
+      locale: locale,
+      bundleName: bundleName,
+      fileName: bundleFile,
+      output: bundlePath,
+      newFile: !existingBundle
+    }
+
+    // no need to regenerate a bundle that has not been modified
+    if (existingBundle && JSON.stringify(existingBundle) === JSON.stringify(bundle)) {
+      bundleData.changed = false;
+      return Promise.resolve(bundleData);
+    }
+
+    return writeFile(bundlePath, data).then(function () {
+      bundleData.changed = true;
+      return Promise.resolve(bundleData);
+    }, function (err) {
+      return Promise.reject(err);
+    });
+  });
+}
+
+function transformLocaleBundles(extension, serializer, output, locales, localeBundles) {
+  return [].concat.apply([], (locales || []).map(function (locale) {
+    const bundles = localeBundles[locale];
+    return transformBundle(extension, serializer, output, locale, bundles);
+  }));
+}
+
+function parseRows(locales, values) {
+  const bundles = {};
+  const localeColumnIndexes = {};
+  const validLocales = locales || [];
+  validLocales.forEach(function (locale) {
+    // setup locales whitelist
+    localeColumnIndexes[locale] = -1;
+  });
+  (values || []).forEach(function (sheetRow, rowIndex) {
```

Comment:
Will doing `const parseRows = (locales, values = []) => {}` work with `undefined` being passed in as `values`? I just ask because [this line](https://github.com/coverhound/i18n-cli/pull/5/files/b0e40ecdfd61fda2eae021fb24e84432c49c1ba4#diff-c94ba88223bcf14c84b36579153756dcR130) always calls this function with two arguments.